src/simd/codepoint_width.zig

```diff
@@ -19,27 +21,41 @@ test "codepointWidth basic" {
     try testing.expectEqual(@as(i8, 2), codepointWidth(0xF900)); // 豈
     try testing.expectEqual(@as(i8, 2), codepointWidth(0x20000)); // 𠀀
     try testing.expectEqual(@as(i8, 2), codepointWidth(0x30000)); // 𠀀
-    // try testing.expectEqual(@as(i8, 1), @import("ziglyph").display_width.codePointWidth(0x100, .half));
+    // const zg = try @import("../../global.zig").Zg.initForTesting();
+    // defer zg.deinitForTesting();
+    // try testing.expectEqual(@as(i8, 1), @import("DisplayWidth").codePointWidth(zg.display_width, 0x100));
 }
 
 // This is not very fast in debug modes, so its commented by default.
 // IMPORTANT: UNCOMMENT THIS WHENEVER MAKING CODEPOINTWIDTH CHANGES.
-// test "codepointWidth matches ziglyph" {
-//     const testing = std.testing;
-//     const ziglyph = @import("ziglyph");
+//test "codepointWidth matches zg" {
+//    const testing = std.testing;
+//    const DisplayWidth = @import("DisplayWidth");
+//    const display_width = try DisplayWidth.init(std.testing.allocator);
+//    defer display_width.deinit(std.testing.allocator);
+//    var success: bool = true;
 //
-//     const min = 0xFF + 1; // start outside ascii
-//     for (min..std.math.maxInt(u21)) |cp| {
-//         const simd = codepointWidth(@intCast(cp));
-//         const zg = ziglyph.display_width.codePointWidth(@intCast(cp), .half);
-//         if (simd != zg) mismatch: {
-//             if (cp == 0x2E3B) {
-//                 try testing.expectEqual(@as(i8, 2), simd);
-//                 break :mismatch;
-//             }
+//    const min = 0xFF + 1; // start outside ascii
+//    for (min..0x110000) |cp| {
+//        const simd = codepointWidth(@intCast(cp));
+//        const zg_width = DisplayWidth.codePointWidth(display_width, @intCast(cp));
+//        if (simd != zg_width) mismatch: {
+//            if (cp == 0x2E3B) {
+//                try testing.expectEqual(@as(i8, 2), simd);
+//                std.log.warn("mismatch for 0x2e3b cp=U+{x} simd={} zg={}", .{ cp, simd, zg_width });
+//                break :mismatch;
+//            }
 //
-//             std.log.warn("mismatch cp=U+{x} simd={} zg={}", .{ cp, simd, zg });
-//             try testing.expect(false);
-//         }
-//     }
-// }
+//            if (cp == 0x890) {
+//                try testing.expectEqual(@as(i8, 0), simd);
+//                try testing.expectEqual(@as(i8, 1), zg_width);
+//                break :mismatch;
+//            }
```

Comment:
there's actually a bunch more failures when i started to try to run this, but in the `unicode-test` exe i verified this PR doesn't change our width calculation.